tests/import_test.go

```diff
@@ -184,6 +186,108 @@ var _ = Describe("[rfe_id:1115][crit:high][vendor:cnv-qe@redhat.com][level:compo
 
 })
 
+var _ = Describe("DataVolume Garbage Collection", func() {
+	var (
+		f        = framework.NewFramework(namespacePrefix)
+		ns       string
+		err      error
+		config   *cdiv1.CDIConfig
+		origSpec *cdiv1.CDIConfigSpec
+	)
+
+	BeforeEach(func() {
+		ns = f.Namespace.Name
+		config, err = f.CdiClient.CdiV1beta1().CDIConfigs().Get(context.TODO(), common.ConfigName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		origSpec = config.Spec.DeepCopy()
+	})
+
+	AfterEach(func() {
+		By("Restoring CDIConfig to original state")
+		err = utils.UpdateCDIConfig(f.CrClient, func(config *cdiv1.CDIConfigSpec) {
+			origSpec.DeepCopyInto(config)
+		})
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() bool {
+			config, err = f.CdiClient.CdiV1beta1().CDIConfigs().Get(context.TODO(), common.ConfigName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return reflect.DeepEqual(config.Spec, *origSpec)
+		}, 30*time.Second, time.Second).Should(BeTrue())
+	})
+
+	DescribeTable("Should", func(annDeleteAfterCompletion string) {
+		By("Set DataVolumeTTLSeconds to 0")
+		ttl := int32(0)
+		err = utils.UpdateCDIConfig(f.CrClient, func(config *cdiv1.CDIConfigSpec) {
+			config.DataVolumeTTLSeconds = &ttl
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		dv := utils.NewDataVolumeWithHTTPImport("gc-test", "100Mi", fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs))
+		dv.Annotations[controller.AnnDeleteAfterCompletion] = annDeleteAfterCompletion
+		err = controllerutil.SetOwnerReference(config, dv, scheme.Scheme)
+		Expect(err).ToNot(HaveOccurred())
+
+		By(fmt.Sprintf("Create new datavolume %s", dv.Name))
+		dv, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, ns, dv)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verify pvc was created")
+		pvc, err := utils.WaitForPVC(f.K8sClient, ns, dv.Name)
+		Expect(err).ToNot(HaveOccurred())
+		f.ForceBindIfWaitForFirstConsumer(pvc)
+
+		if annDeleteAfterCompletion != "false" {
```

Comment:
Instead of having this if/else in here, it might make sense to pass a function into the table. The common information between the two looks like its the datavolume name and namespace. So if you define a 'verifyGCOccured(name, namespace string)` and a `verifyNoGC(name, namespace string)` then you can have that function passed as an argument to to the table, and in the table.Entry pass which function you would like to call to verify. This eliminates the giant if/else in the table and make stuff more readable.

Reply:
fixed